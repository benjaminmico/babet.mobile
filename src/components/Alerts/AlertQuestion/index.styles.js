import styled from 'styled-components'

export const AlertQuestionContainer = styled.View`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0px;
  z-index: 100;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.7);
`
export const AlertQuestionContentContainer = styled.View`
  background-color: ${props => props.backgroundColor};
  border-radius: 12.6px;
  width: 335px;
  height: 176px;
  justify-content: center;
  align-items: center;
`

export const AlertQuestionButtonsContainer = styled.View`
  position: absolute;
  width: 162px;
  left: 150px;
  bottom: 15px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`

export const AlertQuestionText = styled.Text`
  color: ${props => props.color};
  font-family: 'AvenirNext-DemiBold';
  font-size: 15px;
  letter-spacing: -0.16px;
  margin-bottom: 37px;
  padding-left: 20px;
  padding-right: 20px;
  text-align: center;
`
