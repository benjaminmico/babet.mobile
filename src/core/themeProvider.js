import AsyncStorage from '@react-native-community/async-storage'
import React, {useContext, useEffect, useState} from 'react'
import {useDarkMode} from 'react-native-dark-mode'
import THEMES from './themes.json'

const STORAGE_KEY = 'THEME_ID'
const ThemeContext = React.createContext()

export const ThemeContextProvider = ({children}) => {
  const [themeID, setThemeID] = useState()

  const isDarkMode = useDarkMode ? useDarkMode() : false

  useEffect(() => {
    ;(async () => {
      // eslint-disable-next-line no-unused-expressions
      isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
      // const storedThemeID = await AsyncStorage.getItem(STORAGE_KEY)
      // if (storedThemeID) setThemeID(storedThemeID)
      // // eslint-disable-next-line no-unused-expressions
      // else isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
    })()
  }, [])

  return <ThemeContext.Provider value={{themeID, setThemeID}}>{themeID ? children : null}</ThemeContext.Provider>
}

export function withTheme(Component) {
  return props => {
    const {themeID, setThemeID} = useContext(ThemeContext)

    const getTheme = themeID => THEMES.find(theme => theme.key === themeID)
    const setTheme = themeID => {
      AsyncStorage.setItem(STORAGE_KEY, themeID)
      setThemeID(themeID)
    }

    return <Component {...props} themes={THEMES} theme={getTheme(themeID)} setTheme={setTheme} />
  }
}

export function getTheme() {
  const {themeID} = useContext(ThemeContext)

  const getThemeContent = themeID => THEMES.find(theme => theme.key === themeID)

  return getThemeContent(themeID)
}
