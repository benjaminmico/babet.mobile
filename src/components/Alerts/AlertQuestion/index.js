// @flow

import {withTheme} from '@core/themeProvider'
import React from 'react'
import {
  AlertQuestionButtonsContainer,
  AlertQuestionContainer,
  AlertQuestionContentContainer,
  AlertQuestionText,
} from './index.styles'
import Button from '@components/Buttons/Button'

type Props = {
  // question label
  questionLabel: String,
  // cancellable button label
  cancellableButtonLabel: String,
  // cancellable button action
  cancellableButtonPress: () => void,
  // action button label
  actionButtonLabel: String,
  // action button action
  actionButtonPress: () => void,
  // theme
  theme: Object,
}

const AlertQuestion = ({
  questionLabel,
  cancellableButtonLabel,
  cancellableButtonPress,
  actionButtonLabel,
  actionButtonPress,
  theme,
}: Props) => {
  // get theme props
  const {
    colors: {
      components: {backgroundContainer},
      texts: {text: textColor},
    },
  } = theme

  /**
   * render alert question with cancellable && action buttons
   * buttons can only be rendered if concerned labels are passed by props
   * */
  return (
    <AlertQuestionContainer>
      <AlertQuestionContentContainer backgroundColor={backgroundContainer}>
        <AlertQuestionText numberOfLines={3} color={textColor}>
          {questionLabel}
        </AlertQuestionText>
        <AlertQuestionButtonsContainer>
          {cancellableButtonLabel && (
            <Button onPress={cancellableButtonPress} label={cancellableButtonLabel} isCancellable />
          )}
          {actionButtonLabel && <Button onPress={actionButtonPress} label={actionButtonLabel} isPrimaryAction />}
        </AlertQuestionButtonsContainer>
      </AlertQuestionContentContainer>
    </AlertQuestionContainer>
  )
}

export default withTheme(AlertQuestion)
