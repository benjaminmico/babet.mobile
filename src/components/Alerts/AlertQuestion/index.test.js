/**
 * @jest-environment jsdom
 */
import React from 'react'
import {StyleSheet} from 'react-native'
import {ThemeContextProvider} from '@core/themeProvider'
import {mount} from 'enzyme'
import AlertQuestion from '.'

describe.only('AlertQuestion', () => {
  describe('Integrity tests', () => {
    it('renders AlertQuestion correctly', () => {
      const questionLabel = 'Êtes-vous sur de réaliser cette action ?'
      const cancellableButtonLabel = 'Annuler'
      const actionButtonLabel = 'Supprimer'

      const wrapper = mount(
        <ThemeContextProvider>
          <AlertQuestion
            questionLabel={questionLabel}
            cancellableButtonLabel={cancellableButtonLabel}
            actionButtonLabel={actionButtonLabel}
          />
        </ThemeContextProvider>,
      )

      const AlertQuestionText = wrapper.find('Text').first()
      const AlertQuestionCancellableButton = wrapper.find('Button').first()
      const AlertQuestionActionButton = wrapper.find('Button').at(1)
      expect(AlertQuestionText.prop('children')).toBe(questionLabel)
      expect(AlertQuestionCancellableButton.exists()).toBe(true)
      expect(AlertQuestionActionButton.exists()).toBe(true)
    })
  })

  describe('Props tests', () => {
    it('renders AlertQuestion', () => {
      const label = 'Traitement de la photo en cours…'

      const wrapper = mount(
        <ThemeContextProvider>
          <AlertQuestion label={label} />
        </ThemeContextProvider>,
      )

      const AlertQuestionContainer = wrapper.find('View').first()
      const AlertQuestionContentContainer = wrapper.find('View').at(2)
      const AlertQuestionButtonsContainer = wrapper.find('View').at(4)
      const AlertQuestionText = wrapper.find('Text').first()

      expect(wrapper).toMatchSnapshot()

      const stylesContainer = StyleSheet.flatten(AlertQuestionContainer.prop('style'))
      expect(stylesContainer.position).toBe('absolute')
      expect(stylesContainer.top).toBe(0)
      expect(stylesContainer.left).toBe(0)
      expect(stylesContainer.right).toBe(0)
      expect(stylesContainer.bottom).toBe(0)
      expect(stylesContainer.justifyContent).toBe('center')
      expect(stylesContainer.alignItems).toBe('center')

      const stylesContentContainer = StyleSheet.flatten(AlertQuestionContentContainer.prop('style'))
      expect(stylesContentContainer.backgroundColor).toBe('#FFFFFF')
      expect(stylesContentContainer.borderRadius).toBe(12.6)
      expect(stylesContentContainer.width).toBe(335)
      expect(stylesContentContainer.height).toBe(176)
      expect(stylesContentContainer.justifyContent).toBe('center')
      expect(stylesContentContainer.alignItems).toBe('center')

      const stylesButtonsContainer = StyleSheet.flatten(AlertQuestionButtonsContainer.prop('style'))
      expect(stylesButtonsContainer.position).toBe('absolute')
      expect(stylesButtonsContainer.width).toBe(192)
      expect(stylesButtonsContainer.left).toBe(120)
      expect(stylesButtonsContainer.bottom).toBe(15)
      expect(stylesButtonsContainer.flexDirection).toBe('row')
      expect(stylesButtonsContainer.justifyContent).toBe('space-between')
      expect(stylesContentContainer.alignItems).toBe('center')

      const stylesText = StyleSheet.flatten(AlertQuestionText.prop('style'))
      expect(stylesText.color).toBe('#000000')
      expect(stylesText.fontFamily).toBe('AvenirNext-DemiBold')
      expect(stylesText.fontSize).toBe(15)
      expect(stylesText.letterSpacing).toBe(-0.16)
      expect(stylesText.marginBottom).toBe(37)
      expect(stylesText.paddingLeft).toBe(20)
      expect(stylesText.paddingRight).toBe(20)
      expect(stylesText.textAlign).toBe('center')
    })
  })
})
