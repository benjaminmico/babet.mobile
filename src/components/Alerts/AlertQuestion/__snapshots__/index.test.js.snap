// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertQuestion Props tests renders AlertQuestion 1`] = `
<ThemeContextProvider>
  <Component
    label="Traitement de la photo en cours…"
  >
    <AlertQuestion
      label="Traitement de la photo en cours…"
      setTheme={[Function]}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "color": "#101112",
          "colors": Object {
            "components": Object {
              "background": "#FFFFFF",
              "backgroundAction": "#F3F2F2",
              "backgroundContainer": "#FFFFFF",
              "backgroundFilter": "#F3F2F2",
              "backgroundInputButton": "#5100FF",
              "backgroundPrimaryAction": "#5100FF",
              "icon": "#000000",
            },
            "palette": Object {
              "black": "#000000",
              "lost": "#CA1010",
              "pending": "#7C7C7C",
              "purple": "#5100FF",
              "white": "#FFFFFF",
              "won": "#14B916",
            },
            "shadows": Object {},
            "texts": Object {
              "buttons": Object {
                "actions": "#5100FF",
                "cancellable": "#3E3E3E",
                "filter": "#3E3E3E",
                "isPrimaryAction": "#FFFFFF",
              },
              "description": "#ADADAD",
              "filters": Object {
                "multiFilter": "#5100FF",
                "selected": "#000000",
                "unselected": "#A3A2A2",
              },
              "input": Object {
                "label": "#7C7C7C",
                "value": "#000000",
              },
              "labelInput": "#7C7C7C",
              "reverseText": "#FFFFFF",
              "text": "#000000",
              "title": "#000000",
            },
          },
          "isShadow": true,
          "key": "white",
          "statusBar": "dark-content",
        }
      }
      themes={
        Array [
          Object {
            "backgroundColor": "#101112",
            "color": "#ffffff",
            "colors": Object {
              "components": Object {
                "background": "#101112",
                "backgroundAction": "#F3F2F2",
                "backgroundContainer": "#2F3134",
                "backgroundFilter": "#F3F2F2",
                "backgroundInputButton": "#5100FF",
                "backgroundPrimaryAction": "#5100FF",
                "icon": "#FFFFFF",
              },
              "palette": Object {
                "black": "#000000",
                "lost": "#CA1010",
                "pending": "#7C7C7C",
                "purple": "#5100FF",
                "white": "#FFFFFF",
                "won": "#14B916",
              },
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "cancellable": "#3E3E3E",
                  "filter": "#3E3E3E",
                  "isPrimaryAction": "#FFFFFF",
                },
                "description": "#ADADAD",
                "filters": Object {
                  "multiFilter": "#5100FF",
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "input": Object {
                  "label": "#7C7C7C",
                  "value": "#FFFFFF",
                },
                "reverseText": "#FFFFFF",
                "text": "#FFFFFF",
                "title": "#FFFFFF",
              },
            },
            "isShadow": false,
            "key": "dark",
            "statusBar": "light-content",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#101112",
            "colors": Object {
              "components": Object {
                "background": "#FFFFFF",
                "backgroundAction": "#F3F2F2",
                "backgroundContainer": "#FFFFFF",
                "backgroundFilter": "#F3F2F2",
                "backgroundInputButton": "#5100FF",
                "backgroundPrimaryAction": "#5100FF",
                "icon": "#000000",
              },
              "palette": Object {
                "black": "#000000",
                "lost": "#CA1010",
                "pending": "#7C7C7C",
                "purple": "#5100FF",
                "white": "#FFFFFF",
                "won": "#14B916",
              },
              "shadows": Object {},
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "cancellable": "#3E3E3E",
                  "filter": "#3E3E3E",
                  "isPrimaryAction": "#FFFFFF",
                },
                "description": "#ADADAD",
                "filters": Object {
                  "multiFilter": "#5100FF",
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "input": Object {
                  "label": "#7C7C7C",
                  "value": "#000000",
                },
                "labelInput": "#7C7C7C",
                "reverseText": "#FFFFFF",
                "text": "#000000",
                "title": "#000000",
              },
            },
            "isShadow": true,
            "key": "white",
            "statusBar": "dark-content",
          },
        ]
      }
    >
      <Styled(View)>
        <StyledNativeComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "displayName": "Styled(View)",
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  align-items: center;
  justify-content: center;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
            >
              <Styled(View)
                backgroundColor="#FFFFFF"
              >
                <StyledNativeComponent
                  backgroundColor="#FFFFFF"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "displayName": "Styled(View)",
                      "inlineStyle": InlineStyle {
                        "rules": Array [
                          "
  background-color: ",
                          [Function],
                          ";
  border-radius: 12.6px;
  width: 335px;
  height: 176px;
  justify-content: center;
  align-items: center;
",
                        ],
                      },
                      "render": [Function],
                      "styledComponentId": "StyledNativeComponent",
                      "target": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <View
                    backgroundColor="#FFFFFF"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#FFFFFF",
                          "borderRadius": 12.6,
                          "height": 176,
                          "justifyContent": "center",
                          "width": 335,
                        },
                      ]
                    }
                  >
                    <View
                      backgroundColor="#FFFFFF"
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#FFFFFF",
                            "borderRadius": 12.6,
                            "height": 176,
                            "justifyContent": "center",
                            "width": 335,
                          },
                        ]
                      }
                    >
                      <Styled(Text)
                        color="#000000"
                        numberOfLines={3}
                      >
                        <StyledNativeComponent
                          color="#000000"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "displayName": "Styled(Text)",
                              "inlineStyle": InlineStyle {
                                "rules": Array [
                                  "
  color: ",
                                  [Function],
                                  ";
  font-family: 'AvenirNext-DemiBold';
  font-size: 15px;
  letter-spacing: -0.16px;
  margin-bottom: 37px;
  padding-left: 20px;
  padding-right: 20px;
  text-align: center;
",
                                ],
                              },
                              "render": [Function],
                              "styledComponentId": "StyledNativeComponent",
                              "target": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          numberOfLines={3}
                        >
                          <Text
                            color="#000000"
                            numberOfLines={3}
                            style={
                              Array [
                                Object {
                                  "color": "#000000",
                                  "fontFamily": "AvenirNext-DemiBold",
                                  "fontSize": 15,
                                  "letterSpacing": -0.16,
                                  "marginBottom": 37,
                                  "paddingLeft": 20,
                                  "paddingRight": 20,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            <Text
                              color="#000000"
                              numberOfLines={3}
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "AvenirNext-DemiBold",
                                    "fontSize": 15,
                                    "letterSpacing": -0.16,
                                    "marginBottom": 37,
                                    "paddingLeft": 20,
                                    "paddingRight": 20,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                            />
                          </Text>
                        </StyledNativeComponent>
                      </Styled(Text)>
                      <Styled(View)>
                        <StyledNativeComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "displayName": "Styled(View)",
                              "inlineStyle": InlineStyle {
                                "rules": Array [
                                  "
  position: absolute;
  width: 192px;
  left: 120px;
  bottom: 15px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
",
                                ],
                              },
                              "render": [Function],
                              "styledComponentId": "StyledNativeComponent",
                              "target": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "bottom": 15,
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "left": 120,
                                  "position": "absolute",
                                  "width": 192,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "bottom": 15,
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "left": 120,
                                    "position": "absolute",
                                    "width": 192,
                                  },
                                ]
                              }
                            />
                          </View>
                        </StyledNativeComponent>
                      </Styled(View)>
                    </View>
                  </View>
                </StyledNativeComponent>
              </Styled(View)>
            </View>
          </View>
        </StyledNativeComponent>
      </Styled(View)>
    </AlertQuestion>
  </Component>
</ThemeContextProvider>
`;
